src/ng-xform/date-range-field/date-range-field.component.ts

```diff
@@ -0,0 +1,85 @@
+import { DateRangeField } from './../fields/date-range-field';
+import { DatePipe } from '@angular/common';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, AfterContentInit, ElementRef, AfterViewInit, Inject, LOCALE_ID } from '@angular/core';
+import { BsDatepickerConfig } from 'ngx-bootstrap/datepicker';
+
+import { BaseDynamicFieldComponent } from '../field-components/base-dynamic-field.component';
+import { DateField } from '../fields';
+
+/**
+ * Component to generate a bootstrap form field of Date type
+ *
+ * :editing: Flag to control component state
+ * :form: FormGroup containing the field
+ * :field: Intance of field configurations
+ */
+@Component({
+  selector: 'ng-xform-date-range-field',
+  templateUrl: './date-range-field.component.html',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: DateRangeFieldComponent,
+    multi: true
+  }],
+})
+export class DateRangeFieldComponent extends BaseDynamicFieldComponent<DateRangeField> implements AfterViewInit, AfterContentInit,
+  ControlValueAccessor {
+  config: Partial<BsDatepickerConfig>;
+  componentControl = new FormControl();
+  private input: HTMLInputElement;
+
+  _onChange = (value: any) => { };
```

Comment:
Why you need that?

Reply:
To avoid error if it is called before registerOnChange